packages/common/src/otel.ts

```diff
@@ -0,0 +1,46 @@
+import { isSpanContextValid, Span, SpanContext } from '@opentelemetry/api';
+import { defaultDataConverter } from './converter/data-converter';
+import { Headers } from './interceptors';
+
+export const TRACE_HEADER = 'Otel-Trace-Context';
+
+/**
+ * If found, return a span context deserialized from the provided headers
+ */
+export async function extractSpanContextFromHeaders(headers: Headers): Promise<SpanContext | undefined> {
+  const encodedSpanContext = headers[TRACE_HEADER];
+  if (encodedSpanContext == null) {
+    return undefined;
+  }
+  const decoded: SpanContext = await defaultDataConverter.fromPayload(encodedSpanContext);
+  if (isSpanContextValid(decoded)) {
+    return decoded;
+  }
+  return undefined;
+}
+
+/**
+ * Given a span context, insert it into the provided headers if they exist
+ */
+export async function insertSpanContextIntoHeaders(context: SpanContext, headers?: Headers): Promise<void> {
+  if (headers === undefined) {
+    return;
+  }
+  headers[TRACE_HEADER] = await defaultDataConverter.toPayload(context);
+}
+
+/**
+ * Link a span to an maybe-existing span context
+ */
+export function linkSpans(fromSpan: Span, toContext?: SpanContext): void {
+  if (toContext != null) {
+    // TODO: I have to go around typescript because otel api 😢
+    //  See https://github.com/open-telemetry/opentelemetry-js-api/issues/124
+    const links = (fromSpan as any).links;
+    if (links == null) {
```

Comment:
are they actually null or undefined? Prefer to use `===`.

Reply:
My understanding is this is an accepted safe way to check for both at once.

Reply:
https://stackoverflow.com/questions/28975896/is-there-a-way-to-check-for-both-null-and-undefined/28984306#28984306 -- Seems easier than needing to remember which or check for both in various places